### ISSUE TITLE
Migrate EdDSA Private Key to iCloud Keychain

### ISSUE THREAD

**Issue by gregcotten** [sparkle-project/Sparkle#2029]
## Summary

Mac devs will often have multiple Macs for development purposes and often upgrade to the latest/greatest. Because the Sparkle EdDSA key is stored in the login keychain and not iCloud keychain, it is on the burden of the developer to manually copy or move the keychain item from one Mac to the next. There are several GitHub issues from devs whose apps have become completely hosed/bricked because of this.

## Possible Fix

Change the location of the keychain item from `login` to iCloud. I (think?) it is strictly better!

**Comment by zorgiepoo** [comment]
I don't think we can prefer it by default as it may not be available (I only recently started using it personally). It may be nice to add an option to generate_keys, sign_update, generate_appcast to use an alternate keychain though (which has been brought up).

Regular app updates are not hosed by losing an old EdDSA key due to Developer ID signing / key rotation (our docs cover this).

**Comment by gregcotten** [comment]
I have read the docs, yes, but that option (Developer ID signing / key rotation) is a noted as a "last resort" and I'd prefer to never be in a "last resort" situation.

You bring up a good point about iCloud Keychain not being available in some cases - you could fall back to the `login` keychain in this instance. I also wouldn't mind the solution of specifying the keychain to use.

**Comment by zorgiepoo** [comment]
PRs improving this are welcome.

I took a brief look at the keychain APIs and by the looks of it, syncing with iCloud looks do-able, and custom keychain APIs look unsupported. So by that assessment I would just look at iCloud.

This is the relevant key I found that I think we would just need to pass in, and maybe it would automatically all just work, but I haven't tested it - https://developer.apple.com/documentation/security/ksecattrsynchronizable?language=objc

**Comment by zorgiepoo** [comment]
I took a look into this but sadly I do not think this is technically possible and came to same results as https://github.com/jfitzell/mozilla-keychain/issues/62 (or at least I couldn't get it to work at all due to lack of entitlements even when trying to add an access group entitlement).

I have updated the website docs for more clarification on key rotation. Being a "last resort" was not entirely accurate wording, and at least this clarifies that updates through normal app updates will not become hosed, and losing only one pair of keys is not too big of a deal.
